domain/infosync/info.go

```diff
@@ -358,8 +422,33 @@ func (is *InfoSyncer) newSessionAndStoreServerInfo(ctx context.Context, retryCnt
 		err := is.storeServerInfo(ctx)
 		return errors.Trace(err)
 	})
-	err = is.storeServerInfo(ctx)
-	return err
+	return is.storeServerInfo(ctx)
+}
+
+// newTopologySessionAndStoreServerInfo creates a new etcd session and stores server info to etcd.
+func (is *InfoSyncer) newTopologySessionAndStoreServerInfo(ctx context.Context, retryCnt int) error {
+	if is.etcdCli == nil {
+		return nil
+	}
+	logPrefix := fmt.Sprintf("[topology-syncer] %s/%s:%d", TopologyInformationPath, is.info.IP, is.info.Port)
+	session, err := owner.NewSession(ctx, logPrefix, is.etcdCli, retryCnt, TopologySessionTTL)
+	if err != nil {
+		return err
+	}
+
+	is.topologySession = session
+	return is.StoreTopologyInfo(ctx)
+}
+
+// refreshTopology refreshes etcd topology with ttl stored in "/topology/tidb/ip:port/ttl".
+func (is *InfoSyncer) refreshTopology(ctx context.Context) error {
+	if is.etcdCli == nil {
+		return nil
+	}
+	key := fmt.Sprintf("%s/%s:%v/ttl", TopologyInformationPath, is.info.IP, is.info.Port)
+	return util.PutKVToEtcd(ctx, is.etcdCli, keyOpDefaultRetryCnt, key,
+		time.Now().String(),
```

Comment:
Is this a timestamp, or a human readable string?

Reply:
An RFC String

Reply:
I think using timestamp should be better so that you can check the aliveness easily at Dashboard side.

Reply:
Now it's timestamp